python/doc/examples/calibration/bayesian_calibration/plot_rwmh_python_distribution.py

```diff
@@ -0,0 +1,178 @@
+#
+# This method is illustrated on a simple lifetime study test-case, which involves censored data, as described hereafter.
+#
+# In the following, we assume that the lifetime :math:`T_i` of an industrial component follows the Weibull distribution :math:`\mathcal W(\alpha, \beta)`, with CDF  given by :math:`F(t|\alpha,\beta)= 1 - e^{-\left( \frac{t}{\beta} \right)^\alpha}`
+#
+# Our goal is to estimate the model parameters :math:`\alpha, \beta` based on a dataset of recorded failures :math:`(t_1, \ldots, t_n),` some of which correspond to actual failures, and the remaining are right-censored. Let :math:`(f_1, \ldots, f_n) \in \{0,1\}^n` represent the nature of each datum, :math:`f_i=1` if :math:`t_i` corresponds to an actual failure, :math:`f_i=0` if it is right-censored.
+#
+# Note that the likelihood of each recorded failure is given by the Weibull density: :math:`\mathcal L(t_i | f_i=1, \alpha, \beta) = \frac{\alpha}{\beta}\left( \frac{t_i}{\beta} \right)(\alpha-1)e^{-\left( \frac{t_i}{\beta} \right)^\alpha},` while the likelihood of each right-censored observation is given by: :math:`\mathcal L(t_i | f_i=0, \alpha, \beta) = e^{-\left( \frac{t_i}{\beta} \right)^\alpha}.`
+#
+# Furthermore, assume that the prior information available on :math:`\alpha, \beta` is represented by independent prior laws, whose respective densities are denoted by :math:`\pi(\alpha)` and :math:`\pi(\beta).`
+#
+# Then, the posterior distribution of :math:`(\alpha, \beta)`, representing the update of the prior information on :math:`(\alpha, \beta)` given the dataset has a PDF which is proportional to:
+#
+#
+# .. math::
+#   \pi(\alpha, \beta | (t_1, f_1), \ldots, (t_n, f_n) ) \propto \pi(\alpha)\pi(\beta) \left(\frac{\alpha}{\beta}\right)^{\sum_i f_i} \left(\prod_{f_i = 1} \frac{t_i}{\beta}\right)^{\alpha-1} \exp\left[-\sum_{i=1}^n\left(\frac{t_i}{\beta}\right)^\alpha\right].
+#
+# Set up the :class:`PythonDistribution`
+# --------------------------------------
+#
+# The :class:`~openturns.RandomWalkMetropolisHastings` class can be used to sample from the posterior distribution. It relies on the following objects:
+#
+# - The conditional density :math:`p(t_{1:n}|f_{1:n}, \alpha, \beta)` must be defined as a probability distribution
+# - The prior probability density :math:`\pi(\vect{\theta})`, reflecting beliefs about the possible values
+#   of :math:`\vect{\theta} = (\alpha, \beta)` before the experimental data are considered.
+# - Initial values :math:`\vect{\theta}_0` for the calibration parameters
+# - Proposal distributions used to update each parameter sequentially.
+#
+# The censured Weibuill likelihood is outside the usual catalog of probability distributions in OpenTURNS, hence we need to define it using the :class:`~openturns.PythonDistribution` class.
+
+
+# %%
+
+import numpy as np
+import openturns as ot
+from openturns.viewer import View
+ot.Log.Show(ot.Log.NONE)
```

Comment:
```suggestion
ot.Log.Show(ot.Log.NONE)
ot.RandomGenerator.SetSeed(123)
```
Pour assurer la reproductibilité des résultats.